Service/AlgoliaConnector.php

```diff
@@ -453,12 +455,20 @@ function ($object) use ($action) {
     /**
      * @param string $indexName
      * @param array $response
+     * @param int|null $storeId
      * @return void
      */
-    protected static function setLastOperationInfo(string $indexName, array $response): void
+    protected static function setLastOperationInfo(string $indexName, array $response, ?int $storeId = null): void
     {
         self::$lastUsedIndexName = $indexName;
-        self::$lastTaskId= $response[self::ALGOLIA_API_TASK_ID] ?? null;
+        self::$lastTaskId = $response[self::ALGOLIA_API_TASK_ID] ?? null;
+
+        if (!is_null($storeId)) {
+            self::$lastTaskInfoByStore[$storeId] = [
+                'indexName' => $indexName,
```

Comment:
Just a note for future. I understand why this is necessary for this situation but I think this is an example where using the `IndexNameFetcher` might have value to help enforce SRP.

Reply:
Understood, I guess this will be part of the work to do in MAGE-1170 :)